### PR TITLE
ci: [CNI dualstack] Added e2e test pipeline for linux dualstack scenario

### PR DESCRIPTION
**Pull request by tamilmani1989**
**Reason for Change**:

This PR adds pipeline which runs upstream dualstack e2e tests for linux clusters on every PR trigger. This makes sure that there is no regression in dualstack datapath due to that PR.  The kubernetes upstream tests doesn't cover external lb scenario which is azure dependent.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: .pipelines/singletenancy/aks-engine/e2e-step-template.yaml [hunk 1]
```diff
@@ -21,10 +21,10 @@ steps:
       if [ -z '$(AKS_ENGINE_VERSION)' ]
       then
         echo Found set AKS-Engine version '$(AKS_ENGINE_VERSION)'...
-        export aksEVersion=$(curl -L -s -H 'Accept: application/json' https://github.com/Azure/aks-engine/releases/latest | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+        export aksEVersion='$(AKS_ENGINE_VERSION)'
```
Comment (tamilmani1989) [inline .pipelines/singletenancy/aks-engine/e2e-step-template.yaml:24]: @matmerr i guess there is bug here..i fixed it..let me know if im wrong

Comment (tamilmani1989) [reply .pipelines/singletenancy/aks-engine/e2e-step-template.yaml]: not a bug.. but logs are switched

**Comment by tamilmani1989** [comment]
e2e succeeded: https://msazure.visualstudio.com/One/_build/results?buildId=49710545&view=results

File: test/apimodels/cniLinuxDualstack1804.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    "masterProfile": {
+      "count": 1,
+      "dnsPrefix": "duale2e",
+      "vmSize": "Standard_D2_v2"
+    },
+    "agentPoolProfiles": [
+      {
+        "name": "agentdualp1",
+        "count": 3,
```
Comment (matmerr) [inline test/apimodels/cniLinuxDualstack1804.json:41]: can we make this two nodes to cut down on core usage

**Review by vakalapa — APPROVED** [review]
Lgtm,

**Pull request merged**